### ISSUE TITLE
Error: Invalid or incompatible cached data (cachedDataRejected)

### ISSUE THREAD

**Issue by lojoke** [bytenode/bytenode#213]
Sup, I'd like to report a problem that appeared in my project a month ago. I don't know what to do because I've never encountered this error before.

Here is full error code:
    throw new Error('Invalid or incompatible cached data (cachedDataRejected)');
    ^

Error: Invalid or incompatible cached data (cachedDataRejected)
    at Module._extensions.<computed> [as .jsc] ←[90m(C:\Users\User\Desktop\Вбив нитро\vbiv\[4]buy_nitro_on_accounts\←[39mnode_modules\←[4mbytenode←[24m\lib\index.js:275:11←[90m)←[39m
←[90m    at Module.load (node:internal/modules/cjs/loader:1133:32)←[39m
←[90m    at Module._load (node:internal/modules/cjs/loader:972:12)←[39m
←[90m    at Module.require (node:internal/modules/cjs/loader:1157:19)←[39m
←[90m    at require (node:internal/modules/helpers:119:18)←[39m
    at Object.<anonymous> ←[90m(C:\Users\User\Desktop\Вбив нитро\vbiv\[4]buy_nitro_on_accounts\←[39mindex.js:3:5←[90m)←[39m
←[90m    at Module._compile (node:internal/modules/cjs/loader:1275:14)←[39m
←[90m    at Module._extensions..js (node:internal/modules/cjs/loader:1329:10)←[39m
←[90m    at Module.load (node:internal/modules/cjs/loader:1133:32)←[39m
←[90m    at Module._load (node:internal/modules/cjs/loader:972:12)←[39m

Node.js v19.9.0

**Comment by zedd3v** [comment]
Getting the same error

**Comment by janisblaus** [comment]
same issue here! it works fine when compiling tho.

**Comment by luppiale** [comment]
I use electron-builder to build my app from macOs for windows and macOS and I use bytenode to compile main and preload source.
Starting with electron v26 I get the same error on windows machine

**Comment by felipe-linares** [comment]
> I use electron-builder to build my app from macOs for windows and macOS and I use bytenode to compile main and preload source. Starting with electron v26 I get the same error on windows machine

Did you find a solution? We had this working since electron 12, 20, etc. upgrades and just upgraded Electron v30 and it stopped working... kind of a pain to redo our deployment pipelines into a Windows VM so looking to see if you got an easier solution.

**Comment by OsamaAbbas** [comment]
Can you provide a simple example that I can try using `npm install` and `npm start`?

**Comment by felipe-linares** [comment]
@luppiale I actually got it working by downgrading bytenode to 1.4.1.

Since I'm on Electron 30 which runs on Node v20.11 I looked for the oldest commit that supported node v20 and it worked.

Possible some change since then is causing this issue? Maybe intentional? Since I know in other threads you've mentioned that cross-compilation shouldn't be possible @OsamaAbbas

**Comment by OsamaAbbas** [comment]
Glad to hear that downgrading has worked for you.

But still, can you please create a minimal example that reproduce the same issue?

**Comment by thanhle7** [comment]
Same problem with bytenode 1.5.6 on Electron 30.0.2. Downgrading to 28.3.1 solving the problem.

**Comment by SiriffoS** [comment]
Hey! I get the same problem but only some people who download our app seems to get the error when they open it

**Comment by a-parser** [comment]
Right now it depends on the platform, bytecode generated on Linux will not work on Windows/macOS and vice versa

**Comment by SiriffoS** [comment]
One of the people who had this problem has M1Max MBP with Sonoma 14.7.4 and another one didnt mention mackbook version but said he has Sequoia 15.3.1 while I have a M2PRO Ventura 13.4.1, so its the same platform at least?

**Comment by felipe-linares** [comment]
Yeah looks like any form of cross compilation breaks it. Our build pipelines are all done on macOS machines and cross compiling to Windows breaks it.

For context: our application is bundled through webpack and then packaged with electron-builder.

@OsamaAbbas anything we can do to help get more info on this? Or an idea on where we can start looking for the issue.

**Comment by fabi1an** [comment]
![Image](https://github.com/user-attachments/assets/3d59519c-b63a-4724-8d61-0855964a9389)

**Comment by oopses** [comment]
I upgraded from electron 24.8.8 to 37.2.6. Since I also used the same platform to build packages for multiple architectures (for example, building x64 and arm64 on Mac Intel at the same time), an exception occurred after the upgrade. I have seen relevant comments that require the build of different architecture packages to be split into different platforms?

Is it because of the change of V8 engine version?
